src/Interceptor/Interceptor/EventInterceptor.php

```diff
@@ -43,36 +52,43 @@ public function prefix(Instance $instance): Response
      */
     public function suffix(Instance $instance): Response
     {
-        $annotations = $instance->getMethod()
-            ->getAnnotations(Event::class);
-
-        foreach ($annotations as $annotation) {
-            $handler = $this->getHandler(EventHandler::class, $annotation);
+        try {
+            $annotations = $instance->getMethod()
+                ->getAnnotations(Event::class);
+
+            foreach ($annotations as $annotation) {
+                $handler = $this->getHandler(EventHandler::class, $annotation);
+
+                if ($annotation->hasMethod(Event::POST_METHOD) && !$instance->getMethod()->threwException()) {
+                    $event = $handler->make(
+                        $this->getEventName($instance, $annotation, Event::POST_METHOD),
+                        $instance->getMethod()
+                            ->getParameters(),
+                        $instance->getMethod()
+                            ->getReturnedValue(),
+                    );
+                    $handler->send($event);
+                }
+
+                if ($annotation->hasMethod(Event::ON_EXCEPTION_METHOD) && $instance->getMethod()->threwException()) {
+                    $event = $handler->make(
+                        $this->getEventName($instance, $annotation, Event::ON_EXCEPTION_METHOD),
+                        $instance->getMethod()
+                            ->getParameters(),
+                        null,
+                        $instance->getMethod()
+                            ->getException()
+                    );
+                    $handler->send($event);
+                }
 
-            if ($annotation->hasMethod(Event::POST_METHOD) && !$instance->getMethod()->threwException()) {
-                $event = $handler->make(
-                    $this->getEventName($instance, $annotation, Event::POST_METHOD),
-                    $instance->getMethod()
-                        ->getParameters(),
-                    $instance->getMethod()
-                        ->getReturnedValue(),
-                );
-                $handler->send($event);
             }
+        } catch (\LogicException) {
```

Comment:
```$annotations = $instance->getMethod()
                ->getAnnotations(Event::class);```
will throw an exception on use case that does have annotation @Event so we need to catch it here because the generic event doesn't need the annotation to be send